pages/agent/v3/signed_pipelines.md

```diff
@@ -45,86 +47,117 @@ Additionally, we also sign the following aspects of the build for each job:
 
-With regards to specific algorithm choice, basically any of the asymmetric signing algorithms we support are fine and will be secure. If you're not sure which one to use, `EdDSA` is proven to be secure, has a modern design, wasn't designed by a Nation State Actor, and produces nice short signatures.
+For more information on what algorithms the agent supports, run:
 
-For more information on what algorithms the agent supports, run `buildkite-agent tool keygen --help`.
+```sh
+buildkite-agent tool keygen --help
+```
 
-### 2. Making your agents sign steps they upload
+### 2. Configure the agents
 
-Now that your keys are generated, we need to configure your agents to use them. On agents that will be uploading pipelines, add the following to your agent's config file:
+Next, you need to configure your agents to use the keys you generated. On agents that upload pipelines, add the following to the agent's config file:
 
 ```ini
 job-signing-jwks-path=<path to private keyset>
 job-signing-key-id=<the key id you generated earlier>
 job-verification-jwks-path=<path to public keyset>
 ```
 
-This will ensure that whenever those agents upload steps to Buildkite, they'll generate signatures for those steps using the private key we generated earlier. It also ensures that those agents will verify the signatures of any steps they run, using the public key.
+This ensures that whenever those agents upload steps to Buildkite, they'll generate signatures using the private key you generated earlier. It also ensures that those agents verify the signatures of any steps they run, using the public key.
 
-On instances that will verify jobs, add:
+On instances that verify jobs, add:
 
 ```ini
 job-verification-jwks-path=<path to verification keys>
 ```
 
-### 3. Signing the pipeline's configured steps
+### 3. Sign all steps
 
-Builds in Buildkite pipelines are initialized using steps that are configured on the pipeline itself. In the course of running a job, the Buildkite Agent can upload further steps to the pipeline. We will refer to the former as "configured steps" and the latter as "uploaded steps". It's common for the configured steps to be a single step that uploads the remainder of pipeline from [a YAML file in the repository](/docs/pipelines/defining-steps#step-defaults-pipeline-dot-yml-file).
+So far, you've configured agents to sign and verify any steps they upload and run. But there's another source of steps: the Buildkite dashboard. Rather than defining steps in your repository, you can also define steps in a pipeline's settings. The most common use for this is to have a single step that uploads a pipeline definition from [a YAML file in the repository](/docs/pipelines/defining-steps#step-defaults-pipeline-dot-yml-file). These steps should also be signed.
```

Comment:
> Rather than defining steps in your repository, you can also define steps in a pipeline's settings.

this makes it seem a bit like it's an either or, but in 99.99% of cases, it's going to be both, which is why i introduced the terms... they're both of equal importance when it comes to pipelines, both signed and not. 

maybe replacing this paragraph with something like:

```suggestion
So far, you've configured agents to sign and verify any steps they upload and run. However, in Buildkite, steps are defined both in your repo, and in the Buildkite interface itself. The most common use for the steps in buildkite is a single step that uploads a pipeline definition from [a YAML file in the repository](/docs/pipelines/defining-steps#step-defaults-pipeline-dot-yml-file). The steps stored in buildkite will also need to be signed.
```

.... or something, i think there's still some magic that someone better at writing could put on that para 😅

Reply:
Good call. How about this version?

```suggestion
So far, you've configured agents to sign and verify any steps they upload and run. However, you also define steps in a pipeline's settings through the Buildkite dashboard. For example, teams commonly use a single step in the Buildkite dashboard to upload a pipeline definition from [a YAML file in the repository](/docs/pipelines/defining-steps#step-defaults-pipeline-dot-yml-file). These steps should also be signed.
```